src/lib/io/sol-i2c-impl-linux.c

```diff
@@ -903,11 +903,11 @@ sol_i2c_write_register(struct sol_i2c *i2c, uint8_t reg, const uint8_t *values,
 
 #ifdef WORKER_THREAD
     i2c->async.worker = sol_worker_thread_new(&spec);
-    SOL_NULL_CHECK(i2c->async.worker, false);
+    SOL_NULL_CHECK(i2c->async.worker, NULL);
```

Comment:
I suspect there are ohter places in the code where `NULL` is returned instead of `false`, or an `int` is returned instead of a boolean (with the implicit casting and all). I wonder if there's a way to find all occurrences of this.